components/motor/gpio/setup.go

```diff
@@ -42,6 +42,22 @@ type Config struct {
 	TicksPerRotation int            `json:"ticks_per_rotation,omitempty"`
 }
 
+// Validate ensures all parts of the config are valid.
+func (config *Config) Validate(path string) ([]string, error) {
+	var deps []string
+
+	if config.BoardName == "" {
+		return nil, errors.New("motor requires a board")
```

Comment:
use `utils.NewCOnfigFieldRequiredError(path, "board")` for this error and any similar error with a field required in this PR.

That method can be found in the `go.viam.com/utils` package.